ci-operator/step-registry/observers/ocp-baremetal-qe/observers-ocp-baremetal-qe-commands.sh

```diff
@@ -0,0 +1,411 @@
+  if [[ $whatToCheck == "boot" ]]; then
+      if [[ $IS_PXE_JOB = true ]]; then
+          # PXE BOOT
+          # BOOT_IMAGE=/ci-op-07q5cy9i/vmlinuz_x86_64 debug nosplash ip=eno8303:dhcp ip=eno12399np0:off ip=eno12409np1:off console=tty1 console=ttyS0,115200n8
+          if [[ $cmdline == *"$NAMESPACE"* ]]; then
+            echo -e "Booted PXE image version \n $cmdline \n matches Prow namespace $NAMESPACE"
+          else
+            echo -e "Booted PXE image version \n $cmdline \n DOES NOT match Prow namespace $NAMESPACE"
+            handleNode "${bmhost}" "${EXIT_CODE_WRONG_VERSION}"
+          fi
+      else
+          # ISO BOOT
+          # BOOT_IMAGE=/images/pxeboot/vmlinuz coreos.liveiso=rhcos-415.92.202311241643-0 ignition.firstboot ignition.platform.id=metal
+          expected_x86_version=$(yq .architectures.x86_64.artifacts.metal.release $COREOS_STREAM_FILE)
+          echo "Expected x86_64 version to match booted image: $expected_x86_version"
+          expected_arm64_version=$(yq .architectures.aarch64.artifacts.metal.release $COREOS_STREAM_FILE)
+          echo "Expected ARM64 version to match booted image: $expected_arm64_version"
+          if [[ $cmdline == *"$expected_x86_version"* ]] || [[ $cmdline == *"$expected_arm64_version"* ]] ; then
+            echo -e "Booted ISO image version \n $cmdline \n matches expected version \n x86 $expected_x86_version arm64 $expected_arm64_version"
+            # Use code '0' when everything is working as expected, green junit report
+            echo 0 "${host} ${name}" >> "${ARTIFACT_DIR}/install-status.txt"
+          else
+            echo -e "Booted ISO image version $cmdline"
+            echo -e "DOES NOT match expected versions : x86 $expected_x86_version arm64 $expected_arm64_version"
+            handleNode "${bmhost}" "${EXIT_CODE_WRONG_VERSION}"
+          fi
+      fi
+  elif [[ $whatToCheck == "disk" ]]; then
+      echo "$cmdline \n"
+      # BOOT_IMAGE=(hd0,gpt3)/ostree/rhcos-8979e
+      if [[ $cmdline == *"ostree/rhcos"* ]]; then
+        echo -e "Red Hat CoreOS FOUND on disk"
+        journalRecord "${bmhost}" &
+      else
+        echo -e "Red Hat CoreOS NOT FOUND on disk"
+        handleNode "${bmhost}" "${EXIT_CODE_COREOS_NOT_FOUND}"
+      fi
+  fi
+}
+
+function checkNodes(){
+  # shellcheck disable=SC2154
+  for bmhost in $(yq e -o=j -I=0 '.[]' "${HOSTS_FILE}"); do
+    isNodeAlive "${bmhost}" &
+  done
+}
+
+function isPxeJob(){
+  if [[ $JOB_NAME == *"pxe"* ]]; then
+      IS_PXE_JOB=true
+  fi
+  echo "Job name is $JOB_NAME , pxe? $IS_PXE_JOB"
+}
+
+function ipmiRecord(){
+      local bmhost="${1}"
+      . <(echo "$bmhost" | yq e 'to_entries | .[] | (.key + "=\"" + .value + "\"")')
+      echo "SoL recording on ${bmc_address}"
+      ssh "${SSHOPTS[@]}" -t "root@${AUX_HOST}" << EOF > "${ARTIFACT_DIR}/${ip}_${name}_ipmi.txt"
+      ipmitool -I lanplus -H "$bmc_address" -U "$bmc_user" -P "$bmc_pass" -z 8196 sol activate usesolkeepalive &
```

Comment:
Does this work with arm64 HPE nodes?

Reply:
it 'works' but there is no output from the nodes (both arm64 and x86)
https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/47240/rehearse-47240-periodic-ci-openshift-openshift-tests-private-release-4.15-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-f7/1766033600462983168/artifacts/baremetal-compact-agent-ipv4-static-connected-f7/observers-ocp-baremetal-qe/artifacts/192.168.83.75_master-02_ipmi.txt

Reply:
yeah, i don't think we can get output from that for the HPEs.

have a quick look at `cat $(which hpsol)` in the bastion for the HPE nodes.

For the Dell nodes, it is the well-known issue for the agent case, isn't it?

Reply:
for dell nodes, when using ABI it stops working past bootloader screen, but as of now I can't even get the POST logs

Reply:
this is probably due to the fact we are reaching IPMI behind a NAT.

Reply:
I'm running ipmitool from the bastion host, aren't they on the same net segment?

Reply:
oh yes, in that case, you shouldn't have the issue: SOL doesn't work with ipmitool in HPE servers, but Dell servers should still get at least up to grub.

Reply:
have you tried redirecting stderr too? `2>&1`?